lib/src/acter_avatar.dart

```diff
@@ -162,245 +126,242 @@ class _ActerAvatar extends State<ActerAvatar> {
+                      radius: size ?? 24,
+                    ),
+                    Positioned(
+                      left: -7,
+                      bottom: -5,
+                      child: tooltipMsgWithChild(
+                        groupAvatarsInfo[0],
+                        CircleAvatar(
+                          foregroundImage: groupAvatarsInfo[0].avatar,
+                          radius: size ?? 24,
+                        ),
+                      ),
+                    ),
+                    groupAvatarsInfo.length > 1
+                        ? Positioned.fill(
+                            bottom: -5,
+                            child: Align(
+                              alignment: Alignment.bottomRight,
+                              child: Container(
+                                width: 15,
+                                height: 15,
+                                alignment: Alignment.center,
+                                decoration: const BoxDecoration(
+                                  color: Colors.green,
+                                  shape: BoxShape.circle,
+                                ),
+                                child: Text(
+                                  '+${groupAvatarsInfo.length - 1}',
+                                  style: const TextStyle(fontSize: 8),
+                                ),
                               ),
                             ),
-                          ),
-                        )
-                      : const SizedBox.shrink(),
-                ],
-              )
-            : CircleAvatar(
-                foregroundImage: avatar,
-                onForegroundImageError: avatarError,
-                radius: widget.size ?? 24,
-              );
-    }
+                          )
+                        : const SizedBox.shrink(),
+                  ],
+                )
+              : CircleAvatar(
+                  foregroundImage: avatar,
+                  onForegroundImageError: avatarError,
+                  radius: size ?? 24,
+                );
+        }));
   }
 
-  Widget renderSpaceParent(BuildContext context) {
-    if (widget.badgeSize == null) {
-      // nothing. ignore
-      return SizedBox.shrink();
-    }
-    final badgeSize = widget.badgeSize ?? 20;
-    if (widget.avatarsInfo == null || widget.avatarsInfo!.isEmpty) {
-      return SizedBox(
-        height: badgeSize,
-        width: badgeSize,
+  Widget renderBadges(
+    BuildContext context,
+    DefaultAvatarOptions options,
+    double? badgesSize,
+  ) {
+    final size = badgesSize ?? 20;
+    final badges = options.parentBadges!;
+    double badgeOverflow = size / 5;
+    final badgesLength = badges.length;
+    final thresholdCount = badgesLength <= 3 ? badgesLength : 3;
+    List<Widget> children = [];
+
+    for (int i = 0; i < thresholdCount; i++) {
+      children.add(
+        ActerAvatar(options: AvatarOptions(badges[i], size: size)),
```

Comment:
I prefer the briefer
```suggestion
    List<Widget> children = badges.getRange(0, thresholdCount).map((badge) => ActerAvatar(options: AvatarOptions(badge, size: size))).toList(();

```